### PR TITLE
Derive from special-mode instead of fundamental-mode

### PR DESCRIPTION
**Pull request by thierryvolpiatto**
* google-maps-static.el (google-maps-static-refresh): Transform to a
revert-buffer function.
(google-maps-static-mode-map): let's use q and g special-mode bindings.
(google-maps-static-mode): Use special-mode, bind revert-buffer-function.

### CODE REVIEW

File: google-maps-static.el [hunk 1]
```diff
@@ -361,14 +360,14 @@ This function returns the buffer where the map is displayed."
     (define-key map (kbd ",") 'google-maps-static-zoom-out)
     (define-key map (kbd "z") 'google-maps-static-zoom)
     (define-key map (kbd "Z") 'google-maps-static-zoom-remove)
-    (define-key map (kbd "q") 'google-maps-static-quit)
+    ;; (define-key map (kbd "q") 'google-maps-static-quit)
```
Comment (jd) [inline google-maps-static.el]: IIUC you can just delete that.

File: google-maps-static.el [hunk 2]
```diff
@@ -361,14 +360,14 @@ This function returns the buffer where the map is displayed."
...
     (define-key map (kbd "w") 'google-maps-static-copy-url)
     (define-key map (kbd "m") 'google-maps-static-manage-marker)
     (define-key map (kbd "v") 'google-maps-static-manage-visible)
     (define-key map (kbd "c") 'google-maps-static-center)
     (define-key map (kbd "C") 'google-maps-static-center-remove)
     (define-key map (kbd "t") 'google-maps-static-set-maptype)
-    (define-key map (kbd "g") 'google-maps-static-refresh)
+    ;; (define-key map (kbd "g") 'google-maps-static-refresh)
```
Comment (jd) [inline google-maps-static.el]: IIUC you can just delete that and the function that goes with it.

**Comment by thierryvolpiatto** [comment]
Julien Danjou <notifications@github.com> writes:

> @jd requested changes on this pull request.
>
> ------------------------------------------------------------------------------------------------------------------------------------------------------------
>
> In google-maps-static.el:
>
>> @@ -361,14 +360,14 @@ This function returns the buffer where the map is displayed."
>      (define-key map (kbd ",") 'google-maps-static-zoom-out)
>      (define-key map (kbd "z") 'google-maps-static-zoom)
>      (define-key map (kbd "Z") 'google-maps-static-zoom-remove)
> -    (define-key map (kbd "q") 'google-maps-static-quit)
> +    ;; (define-key map (kbd "q") 'google-maps-static-quit)
>
> IIUC you can just delete that.

Done.

> ------------------------------------------------------------------------------------------------------------------------------------------------------------
>
> In google-maps-static.el:
>
>>      (define-key map (kbd "w") 'google-maps-static-copy-url)
>      (define-key map (kbd "m") 'google-maps-static-manage-marker)
>      (define-key map (kbd "v") 'google-maps-static-manage-visible)
>      (define-key map (kbd "c") 'google-maps-static-center)
>      (define-key map (kbd "C") 'google-maps-static-center-remove)
>      (define-key map (kbd "t") 'google-maps-static-set-maptype)
> -    (define-key map (kbd "g") 'google-maps-static-refresh)
> +    ;; (define-key map (kbd "g") 'google-maps-static-refresh)
>
> IIUC you can just delete that and the function that goes with it.

I have removed the commented line, however the function (modified) is
still used as revert-buffer-function.

-- 
Thierry

**Comment by jd** [comment]
I was talking about `google-maps-static-quit` to remove, maybe I commented on the wrong line sorry :)

**Comment by thierryvolpiatto** [comment]
Julien Danjou <notifications@github.com> writes:

> I was talking about google-maps-static-quit to remove, maybe I
> commented on the wrong line sorry :)

Yes, of course, forgot about this one :-), done thanks.

>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.*


-- 
Thierry

**Pull request merged**